### DISCUSSION TITLE
ScriptReference/ContactPoint2D-normalImpulse

### DISCUSSION THREAD

**Discussion by giscus** [Developer-Notes-Extension/Unity-Notes#639] [Notes]
# ScriptReference/ContactPoint2D-normalImpulse

https://docs.unity3d.com/ScriptReference/ContactPoint2D-normalImpulse.html

**Comment by qwe321qwe321qwe321** [comment]
This value is pretty unreliable. Don't use it as the precise physics measure.
Sometimes it returns NaN. 
And most of the time it does not adhere Newton's third law of motion ([Example](https://forum.unity.com/threads/the-value-of-contact2d-normalimpulse-is-wrong-2018-4-11f1.761195/)).
